cloud-gateway-service/src/test/java/org/zowe/apiml/cloudgatewayservice/filters/MissingHeaderRoutePredicateFactoryTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.cloudgatewayservice.filters;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.springframework.mock.http.server.reactive.MockServerHttpRequest;
+import org.springframework.mock.web.server.MockServerWebExchange;
+import org.springframework.web.server.ServerWebExchange;
+
+import java.util.function.Predicate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MissingHeaderRoutePredicateFactoryTest {
+
+    private Predicate<ServerWebExchange> predicate;
+
+    @BeforeEach
+    void setup() {
+        MissingHeaderRoutePredicateFactory.Config config = new MissingHeaderRoutePredicateFactory.Config();
+        config.setHeader("myheader");
+        predicate = new MissingHeaderRoutePredicateFactory().apply(config);
+    }
+
+    @ParameterizedTest(name = "When header {1} is set to {2} and the filter works with header {0} the value after processing should be {3}")
```

Comment:
I can see only 3 items in CSV, are these indexes correct?

Reply:
Good catch, I forgot to update description.